test/unit/anchore_engine/services/apiext/api/helpers/test_images.py

```diff
@@ -0,0 +1,314 @@
+import pytest
+
+from anchore_engine.services.apiext.api.helpers.image_content_response import (
+    _build_os_response,
+    _build_npm_response,
+    _build_gem_response,
+    _build_python_response,
+    _build_java_response,
+    _build_default_response,
+)
+
+
+class TestBuildOsResponse:
+    @pytest.fixture
+    def content_data_entry(self):
+        return {
+            "arch": "amd64",
+            "license": "GPLv2+",
+            "origin": "APT Development Team <deity@lists.debian.org> (maintainer)",
+            "release": "N/A",
+            "size": "4064000",
+            "sourcepkg": "apt-1.8.2",
+            "type": "dpkg",
+            "version": "1.8.2",
+        }
+
+    @pytest.fixture
+    def muli_license_content_data(self, content_data_entry):
+        response = content_data_entry.copy()
```

Comment:
Does it need to copy here? The default for the fixture will be to create the dictionary for every test, just like `setup()` would. Did you experience a different behavior?

Reply:
This is in case both fixtures are used in the same test, and depending on the load order for the test, one fixture would affect the other. I admit I didn't try that out though, and copied just for safety --I'll give it a shot and report back.

Reply:
yup, the copy is needed for safety. Though, currently I'm not using both fixtures in a single test, in case that ever ends up being the case, they are safely decoupled with the `copy()` statement:

```python
   
 # note: "muli_license_content_data" is not using copy()

    def test_thing(self, content_data_entry, muli_license_content_data):
        assert content_data_entry['license'] == 'MIT'
        assert muli_license_content_data['license'] == "MIT GPLv2"
...
```

results in:
```
...
    def test_thing(self, content_data_entry, muli_license_content_data):
>       assert content_data_entry['license'] == 'MIT'
E       AssertionError: assert 'MIT GPLv2' == 'MIT'
E         - MIT
E         + MIT GPLv2

test/unit/anchore_engine/services/apiext/api/helpers/test_images.py:133: AssertionError
```

Reply:
Ah yes, you can't mix and match a fixture that uses another fixture, and then request that same fixture expecting it to be the same. This is because the `return {...}` is already associated with the fixture name, making it effectively a variable value that has already been computer. 

When that is _not the case_, the fixtures don't need to do a copy at all. Using this class for example:

```python
class TestFixture:

    @pytest.fixture
    def obj(self):
        return {'key': 'value'}

    def test_mangle_obj(self, obj):
         assert  obj['key'] == 'value'
         assert obj.get('other') is None
         obj['key'] = 1
         obj['other'] = 2

    def test_mangle_obj_2(self, obj):
         assert  obj['key'] == 'value'
         assert obj.get('other') is None
         obj['key'] = 1
         obj['other'] = 2

    def test_mangle_obj_3(self, obj):
         assert  obj['key'] == 'value'
         assert obj.get('other') is None
         obj['key'] = 1
         obj['other'] = 2
```

Does not produce any failures:

```text
collecting ... collected 3 items

test_common.py::TestFixture::test_mangle_obj PASSED                                                                                                                                                                                                                      [ 33%]
test_common.py::TestFixture::test_mangle_obj_2 PASSED                                                                                                                                                                                                                    [ 66%]
test_common.py::TestFixture::test_mangle_obj_3 PASSED                                                                                                                                                                                                                    [100%]
```